hips/tiles/tile.py

```diff
@@ -95,7 +95,7 @@ def read(cls, meta: HipsTileMeta, filename: str = None) -> 'HipsTile':
 
         if meta.file_format == 'fits':
             hdu_list = fits.open(str(path))
-            data = hdu_list[0].data
+            data = hdu_list[0].data.astype('float')
```

Comment:
Is this `dtype` conversion here really needed? Is it a good idea?
Naively I would think that we should just process data in the dtype given to us, i.e. remove the `.astype('float')` here.
If dtype conversions are needed, then we should have a logic about it, i.e. please leave a short comment why it's done here.